docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+backend pipeline on a single machine. We can use these as minimum requirements:
+
+- Linux server with 4x H100 GPUs
+
+### Service-Oriented
+
+While considering the future, we must understand each targeted persona and
+design services that are focused on their needs.
+
+We should consider the persona of an InstructLab Model Developer and what
+actions and information they care about. That should drive the service
+interface provided by our Backend Service. Some examples of this point:
+
+- An InstructLab Model Developer cares about defining inputs to our pipeline,
+  launching that pipeline, following its status, and consuming its outputs.
+
+- An InstructLab Model Developer does not care about the exact details of how
+  these processes are running on a cluster. They should not require any access
+  to Kubernetes APIs. The cluster itself is the concern of the Cluster
+  Administrator.  Ensuring the Backend Service is configured and running
+  correctly on the cluster is the concern of the Backend Service Owner.
+
+If we map this back to the InstructLab project itself, you can imagine that the
+Backend Maintainers team serves as Backend Service Owners. There may be another
+team, the taxonomy maintainers, that represent InstructLab Model Developers,
+focused on evaluating taxonomy contributions and running batches of them through
+the backend pipeline.
+
+![High-Level Service Diagram](./images/backend-service-high-level.png)
+
+### Observable
+
+The Backend Service must be observable, both by the Backend Service Owners and
+by InstructLab Model Developers.
+
+### Reusable
+
+The Backend Service was envisioned from the beginning as a pipeline of distinct
+steps (data generation, training, evaluation, and publishing). This separation
+of concerns should be maintained in a future service-oriented architecture.
+The diagram under [Service-Oriented](#service-oriented) shows a high-level diagram
+with a single "InstructLab Service API." Implementation of each sub-section of
+the functionality should be done in such a way that the core of it is reusable.
+In some cases that may be via libraries. In others, it may make sense to build a
+microservice that can be used in other contexts.
+
+Within InstructLab, we could have a top-level application that implements the
+entire backend pipeline workflow and does so by making use of each of these
+libraries and/or services.
+
+![Service Breakdown](./images/backend-service-breakdown.png)
+
+### Small Iterations
+
+While we may cast a vision for the future, we desire to get there via a series
+of small iterations. We have a working backend pipeline already and will
+continue to run it every week. In parallel, we can work to evolve it through a
+series of small iterations that evolve our processes over time to use new
+service-based interfaces, gradually evolving the execution process that occurs
+each week.
+
+We do not want a new effort happening in parallel to what is already working
```

Comment:
I agree with this. Is it worth making this point explicit to say that the architectural changes to support the backend service implementation must be sufficiently modular that they can be adopted in the product piece-by-piece? I think this is what the `Small Iterations` implies, but it could also just mean "build the backend one piece at a time" versus "make the backend consumable on piece at a time."